projects/js-packages/licensing/components/activation-screen-controls/index.jsx

```diff
@@ -151,9 +151,20 @@ const ActivationScreenControls = props => {
 		jetpackAnalytics.tracks.recordEvent( 'jetpack_wpa_license_key_activation_view' );
 	}, [] );
 
-	const className = hasLicenseError
-		? 'jp-license-activation-screen-controls--license-field-with-error'
-		: 'jp-license-activation-screen-controls--license-field';
+	const errorTypeMatch = licenseError?.match( /\[[a-z_]+\]/ );
+	const errorType = errorTypeMatch && errorTypeMatch[ 0 ];
+
+	const { ACTIVE_ON_SAME_SITE } = LICENSE_ERRORS;
+	const isLicenseAlreadyAttached = ACTIVE_ON_SAME_SITE === errorType;
+	const className = useMemo( () => {
+		if ( hasLicenseError ) {
+			if ( isLicenseAlreadyAttached ) {
+				return 'jp-license-activation-screen-controls--license-field-with-success';
+			}
+			return 'jp-license-activation-screen-controls--license-field-with-error';
+		}
+		return 'jp-license-activation-screen-controls--license-field';
```

Comment:
This could be flattened a bit
```suggestion
                if ( ! hasLicenseError ) {
                         return 'jp-license-activation-screen-controls--license-field';
                 }

		if ( isLicenseAlreadyAttached ) {
			return 'jp-license-activation-screen-controls--license-field-with-success';
		}

		return 'jp-license-activation-screen-controls--license-field-with-error';
```